File: 백준/Silver/1920. 수 찾기/수 찾기.swift
import Foundation

let N = Int(readLine()!)!
var arr = readLine()!.split(separator: " ").map { Int($0)! }
arr.sort()

let M = Int(readLine()!)!
let targets = readLine()!.split(separator: " ").map { Int($0)! }

for x in targets {
    var l = 0
    var r = arr.count - 1
    var flag = false
    while l <= r {
        var mid = (l + r) / 2
        
        if arr[mid] == x { 
            flag = true
            break
        } else if arr[mid] < x {
            l = mid + 1 
        } else {
            r = mid - 1
        }
    }
    print(flag ? 1 : 0)
}

File: 백준/Silver/1920. 수 찾기/수 찾기.py
n = int(input())
arr1 = list(map(int, input().split()))
arr1.sort()

m = int(input())
arr2 = list(map(int, input().split()))

for x in arr2:
    left = 0
    right = len(arr1) - 1
    flag = False
    
    while left <= right:
        mid = (left + right) // 2
        
        if x == arr1[mid]:
            flag = True
            break
        elif x > arr1[mid]:
            left = mid + 1
        else:
            right = mid - 1
    
    if flag:
        print(1)
    else:
        print(0)
        